test/tap/tests/mysql-protocol_compression_level-t.cpp

```diff
@@ -144,10 +152,16 @@ int main(int argc, char** argv) {
 		goto cleanup;
 	}
 
-	diff = time_proxy_compressed - time_mysql_compressed;
-	performance_diff = (diff * 100) / time_mysql_compressed;
+	time_mysql_without_compressed = calculate_query_execution_time(mysql, query);
```

Comment:
Hi @yashwantsahu20,

why this `ok` has been turned into a `diag`? Measuring the performance difference between compress/not-compress was one of the motivations for the test right? Also, I consider the test output interesting, this is from CI run #134:

```
[2024-12-13 06:12:54] DEBUG    [proxysql-tester.py:695] msg: Row count: 39690000
[2024-12-13 06:12:54] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:12:54.023576  Time taken for query with proxysql without compression: 111327049
[2024-12-13 06:14:46] DEBUG    [proxysql-tester.py:695] msg: Row count: 39690000
[2024-12-13 06:14:46] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:14:46.266343  Time taken for query with proxysql with compression: 112242706
[2024-12-13 06:14:46] DEBUG    [proxysql-tester.py:695] msg: ok 1 - 2024-12-13 06:14:46.266369 - proxysql with compression performed well compared to without compression. Performance difference: 0 percentage
[2024-12-13 06:15:18] DEBUG    [proxysql-tester.py:695] msg: Row count: 39690000
[2024-12-13 06:15:18] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:15:18.532251  Time taken for query with mysql with compression: 32265856
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: Row count: 39690000
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:15:49.671693  Time taken for query with mysql without compression: 31139396
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:15:49.671718  Time difference for mysql, compression and without compression is: -313005196
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: ok 2 - 2024-12-13 06:15:49.709818 - Run-time default compression level is correct: 3
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: ok 3 - 2024-12-13 06:15:49.720507 - Default compression level is correct: 3
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: ok 4 - 2024-12-13 06:15:49.737985 - Run-time Compression level is set correctly: 8
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: ok 5 - 2024-12-13 06:15:49.747345 - Compression level is set correctly: 8
[2024-12-13 06:17:42] DEBUG    [proxysql-tester.py:695] msg: Row count: 39690000
[2024-12-13 06:17:42] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:17:42.583583  Time taken for query with proxysql with compression level 8: 112836184
[2024-12-13 06:17:42] DEBUG    [proxysql-tester.py:695] msg: ok 6 - 2024-12-13 06:17:42.614726 - Run-time Compression level set correctly: 3
[2024-12-13 06:17:42] DEBUG    [proxysql-tester.py:695] msg: ok 7 - 2024-12-13 06:17:42.617790 - Compression level set correctly: 3
[2024-12-13 06:17:42] DEBUG    [proxysql-tester.py:695] msg: Test took 6 min 40.62 sec
```

The value for the test that was commented doesn't make a lot of sense here:

```
[2024-12-13 06:15:49] DEBUG    [proxysql-tester.py:695] msg: # 2024-12-13 06:15:49.671718  Time difference for mysql, compression and without compression is: -313005196
```

I think an `abs` may be missing in the computation. As a side note, maybe make the values reported for this tests floats, as with integers resolution is lost and the output may look like a failure:

```
[2024-12-13 06:14:46] DEBUG    [proxysql-tester.py:695] msg: ok 1 - 2024-12-13 06:14:46.266369 - proxysql with compression performed well compared to without compression. Performance difference: 0 percentage
```

Reply:
Thanksfor your comment.I just thought to remove the test as it was for direct mysql.
Now I have added it back and fixed test value also.